keyvault/client/src/main/java/com/azure/keyvault/models/DeletedSecret.java

```diff
@@ -0,0 +1,77 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.keyvault.models;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+import java.time.Instant;
+import java.time.OffsetDateTime;
+import java.time.ZoneOffset;
+
+public class DeletedSecret extends SecretInfo {
+
+    /**
+     * The url of the recovery object, used to identify and recover the deleted
+     * secret.
+     */
+    @JsonProperty(value = "recoveryId")
+    private String recoveryId;
+
+    /**
+     * The time when the secret is scheduled to be purged, in UTC.
+     */
+    @JsonProperty(value = "scheduledPurgeDate", access = JsonProperty.Access.WRITE_ONLY)
+    private Long scheduledPurgeDate;
```

Comment:
I wonder if rather than store as an intermediate long value if you could deserialize straight into an OffsetDateTime here?

Reply:
Updated with custom unpacking and direct conversion to OffsetDateTime.
No more intermediate Long values.